aiohttp/web.py

```diff
@@ -501,11 +502,15 @@ def run_app(
     except (GracefulExit, KeyboardInterrupt):  # pragma: no cover
         pass
     finally:
-        _cancel_tasks({main_task}, loop)
-        _cancel_tasks(asyncio.all_tasks(loop), loop)
-        loop.run_until_complete(loop.shutdown_asyncgens())
-        loop.close()
-        asyncio.set_event_loop(None)
+        try:
+            main_task.cancel()
+            with suppress(asyncio.CancelledError):
```

Comment:
I think its ok to suppress `CancelledError` here. I couldn't think of a reason it would be a problem.

Reply:
We were already suppressing CancelledError, this just avoids the .call_exception_handler() in _cancel_tasks().